### PR TITLE
[polyhook2] Update port

### PR DESCRIPTION
**Pull request by xeropresence**
Updates polyhook_2 vcpkg to the latest version. 
New version adds functionality to adjust logging verbosity during runtime.

- What does your PR fix? Fixes issue #

- Which triplets are supported/not supported? Have you updated the CI baseline?
x86-windows,x86-windows-static,x64-windows,x64-windows-static,
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes?

### CODE REVIEW

**Comment by LilyWangL** [comment]
@xeropresence Thanks for your PR! Can you please correct the Version field in the CONTROL file? You can update `Version: 2020-02-03` to new version. This is [maintainer-guide.md](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md) documention. You can get more information from it.

**Comment by xeropresence** [comment]
Updated, sorry I missed that when I did the initial commit.

**Comment by xeropresence** [comment]
Is there anything blocking this from being merged?

With the merge of https://github.com/microsoft/vcpkg/pull/8426 the feature could now be enabled with polyhook2, would it make sense to update this pr further or issue a new one?

File: ports/polyhook2/portfile.cmake [hunk 1]
```diff
@@ -40,4 +40,4 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
 endif()
 
 # Handle copyright
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
\ No newline at end of file
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (LilyWangL) [inline ports/polyhook2/portfile.cmake]: what happened here? If nothing change, please remove this change here.

Thanks.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/polyhook2/portfile.cmake (hunk 1 above)
Comment (xeropresence) [reply ports/polyhook2/portfile.cmake]: Githubs web editor automatically inserts a newline at the end of the file if one does not exist. There does not seem to be a way to delete it as it just produces empty commits.

**Comment by strega-nil** [comment]
Cool, thanks @xeropresence :)

**Pull request merged**